src/app/core/services/phase.service.ts

```diff
@@ -20,83 +27,91 @@ export class PhaseService {
     'phase2': 'Team\'s Response Phase',
     'phase3': 'Moderation Phase',
   };
-  private phaseNum: string;
-
-  constructor(private http: HttpClient,
-              private github: GithubService,
-              private errorHandlingService: ErrorHandlingService) {}
-
-  parseEncodedPhase(encodedText: String): string[] {
-    const phase = encodedText.split('@', 4);
-    const moduleOrg = phase[0];
-    const phaseOneUrl = phase[1].split('=', 2)[1];
-    const phaseTwoUrl = phase[2].split('=', 2)[1];
-    const phaseThreeUrl = phase[3].split('=', 2)[1];
-
-    let separator = phaseOneUrl.lastIndexOf('/');
-    const repoName = phaseOneUrl.substring(separator + 1);
+  public sessionData: SessionData;
 
-    let separatorOrg = phaseOneUrl.indexOf('.com');
-    const orgName = phaseOneUrl.substring(separatorOrg + 5, separator);
-
-    separator = phaseTwoUrl.lastIndexOf('/');
-    const repoNameSecond = phaseTwoUrl.substring(separator + 1);
-    separatorOrg = phaseTwoUrl.indexOf('.com');
-    const orgNameSecond = phaseTwoUrl.substring(separatorOrg + 5, separator);
-
-    separator = phaseThreeUrl.lastIndexOf('/');
-    const repoNameThird = phaseThreeUrl.substring(separator + 1);
-    separatorOrg = phaseThreeUrl.indexOf('.com');
-    const orgNameThird = phaseThreeUrl.substring(separatorOrg + 5, separator);
+  private phaseRepoOwners = {
+    phase1: '',
+    phase2: '',
+    phase3: '',
+  };
 
-    return new Array(repoName, orgName, repoNameSecond, orgNameSecond, repoNameThird, orgNameThird, moduleOrg);
+  constructor(private http: HttpClient,
+              private githubService: GithubService,
+              private labelService: LabelService) {}
+
+  /**
+   * Stores the location of the repositories belonging to
+   * each phase of the application.
+   * @param org - name of organization.
+   * @param user - name of user.
+   */
+  setPhaseOwners(org: string, user: string): void {
+    this.orgName = org;
+    this.phaseRepoOwners.phase1 = user;
+    this.phaseRepoOwners.phase2 = org;
+    this.phaseRepoOwners.phase3 = org;
   }
 
-  checkIfReposAccessible(array: any): any {
+  /**
+   * Returns the name of the owner of the input phase.
```

Comment:
Not sure what input phase means?

Reply:
The parameter. Shall I change it to `owner of a given phase`?

Reply:
Yes, makes more sense